packages/wrangler/src/parse.ts

```diff
@@ -0,0 +1,178 @@
+  try {
+    return TOML.parse(input);
+  } catch (err) {
+    const { name, message, line, col } = err;
+    if (name !== "TomlError") {
+      throw err;
+    }
+    // Errors from the TOML parser are formatted as "... at row 1, col 3".
+    // Since the formatter already formats the line and column, the message can be truncated.
+    const text = message.split(" at row ", 2)[0];
+    const lineText = input.split("\n")[line];
+    const location = { file, lineText, line: line + 1, column: col - 2 };
+    throw new ParseError({ text, location });
+  }
+}
+
+/**
+ * A wrapper around `JSON.parse` that throws a friendly error.
+ */
+export function parseJSON(input: string, file?: string): any {
+  try {
+    return JSON.parse(input);
+  } catch (err) {
+    const { message } = err;
+    const index = message.lastIndexOf(" in JSON at position "); // length = 21
+    if (index < 0) {
+      throw err;
+    }
+    const text = message.substring(0, index);
+    const position = parseInt(message.substring(index + 21));
+    const location = indexLocation(input, position, { file });
+    throw new ParseError({ text, location });
+  }
+}
+
+/**
+ * Reads a file and parses it based on its type.
+ */
+export async function readFile(
+  file: string,
+  type?: "json" | "toml"
+): Promise<any> {
+  let content;
+  try {
+    content = await readFile_(file, { encoding: "utf-8" });
+  } catch (err) {
+    // TODO: cleanup common Node.js errors that can be confusing (e.g. ENOENT)
+    throw new ParseError({
+      text: err.message,
+      location: {
+        file,
+      },
+    });
+  }
+  if (type === "json") {
+    return parseJSON(content, file);
+  }
+  if (type === "toml") {
+    return parseTOML(content, file);
+  }
+  return content;
+}
+
+/**
+ * Calculates the line and column location from an index.
+ */
+export function indexLocation(
+  input: string,
+  index: number,
+  extra?: Partial<Location>
+): Partial<Location> {
+  let lineText,
+    line = 0,
+    column = 0,
+    cursor = 0;
+  for (const content of input.split("\n")) {
+    line++;
+    cursor += content.length + 1;
+    if (cursor >= index) {
+      lineText = content;
+      column = content.length - (cursor - index);
+      break;
+    }
+  }
+  return { lineText, line, column, ...extra };
+}
+
+/**
+ * Guesses the line and column location of a search query.
+ */
+export function searchLocation(
+  input: string,
+  search: string,
+  extra?: Partial<Location>
+): Partial<Location> {
+  let lineText,
+    length,
+    line = 0,
+    column = 0;
+  for (const content of input.split("\n")) {
+    line++;
+    const index = content.indexOf(search);
+    if (index >= 0) {
+      lineText = content;
+      column = index;
+      length = search.length;
+      break;
+    }
+  }
+  return { lineText, line, column, length, ...extra };
+}
+
+process.on("uncaughtException", (err) => {
+  if (err instanceof ParseError) {
+    printMessage(err.detail, "error");
+  }
+});
```

Comment:
I am not sure we should have this hidden down in this file.
If we are going to add process level event handlers, then we should be doing it in the top level `index.ts` where the `main()` function is.

Reply:
But, I am pretty sure that this will never happen in production, since we wrap the call to `main()` in a try-catch block.

Reply:
I was using this while testing, will remove.